drivers/serial/uart_stm32.c

```diff
@@ -59,6 +59,15 @@ LOG_MODULE_REGISTER(uart_stm32, CONFIG_UART_LOG_LEVEL);
 
 #define HAS_LPUART DT_HAS_COMPAT_STATUS_OKAY(st_stm32_lpuart)
 
+/* Helper for checking if we can use hardware receive timeouts
+ * instead of the work queue
+ */
+#ifdef CONFIG_SOC_SERIES_STM32U5X
```

Comment:
I'm not quite fond of using series Kconfig directly in files.
I'd prefer you add a Kconfig for the feature and protect it this way, then series that support this feature can be added directly in the Kconfig file rather than having a `#idef` OR'ed list of series poping up in the driver code.

Reply:
Changed the check now. Hopefully this is a good way of doing it?